vm/message/src/lib.rs

```diff
@@ -30,8 +31,16 @@ pub trait Message {
     fn params(&self) -> &Serialized;
     /// gas_price returns gas price for the message
     fn gas_price(&self) -> &TokenAmount;
+    //sets the gas price
+    fn set_gas_price(&mut self, _: TokenAmount);
+    //sets the gas limit
+    fn set_gas_limit(&mut self, _: u64);
+    /// sets a new sequence to the message
+    fn set_sequence(&mut self, _: u64);
     /// Returns the gas limit for the message
     fn gas_limit(&self) -> u64;
     /// Returns the required funds for the message
     fn required_funds(&self) -> TokenAmount;
+    //turns message into cid
+    fn to_cid(&self) -> Result<Cid, String>;
```

Comment:
You don't need this, it's in the `Cbor` trait and has nothing to do with a message type

Reply:
If you look at ChainMessage in lotus, they have a function from switching method of converting to cid. 

https://github.com/filecoin-project/lotus/blob/9b1d2f4c137433c1fe1f5ed44ba2e70105a409f0/chain/types/message.go

Or I could introduce a new type that holds a BoxMessage and also is an implementer of cbor? It just felt like this mapped to what our Message Trait was doing

Reply:
They only do that because it's idiomatic go and there is no way to combine interfaces, for us we would never need this. Better just to add the constraint that the type is also `Cbor` and you don't need this duplicate function attached to a trait that logically doesn't fit